### PR TITLE
Better column counting in case of errors

### PR DESCRIPTION
**Pull request by TimothyGu**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,14 +58,38 @@ Lexer.prototype = {
...
+        allowReturnOutsideFunction: true
+      });
     } catch (ex) {
-      var msg = 'Syntax Error: ' + ex.message;
-      this.error('SYNTAX_ERROR', msg);
+      try {
+        acorn.parse(src, {
+          ecmaVersion: 5,
+          allowReturnOutsideFunction: true
+        });
```
Comment (ForbesLindesay) [inline index.js]: I don't think we need to add the `ecmaVersion: 5` attempt here.  It's only a very limited set of things that fail to parse in `ecmaVersion: 6` and they're very bad things to write.  Lets force `ecmaVersion: 6` while we're publishing a breaking version anyway.

File: index.js [hunk 2]
```diff
@@ -57,14 +58,38 @@ Lexer.prototype = {
...
-  assertExpression: function (exp) {
+  assertExpression: function (exp, noThrow) {
     //this verifies that a JavaScript expression is valid
+    var src = 'return (' + exp + ')';
     try {
-      Function('', 'return (' + exp + ')');
+      acorn.parse(src, {
+        ecmaVersion: 6,
+        allowReturnOutsideFunction: true
+      });
```
Comment (ForbesLindesay) [inline index.js]: We could go one step further by asserting that acorn.parse returns exactly one statement of type return statement.  That prevents things like `true)console.log('oops')(` from being seen as a valid expression.

File: index.js (hunk 1 above)
Comment (TimothyGu) [inline index.js]: Okay.

File: index.js (hunk 2 above)
Comment (TimothyGu) [inline index.js]: Fixed by using my solution in ternjs/acorn#353.

**Pull request merged**